### PR TITLE
[T3A2-3][T1-04]Dao Han

### PR DESCRIPTION
**Pull request by DaoHanLim**
-Changed parser commands to class-level commands
-Added sequence number for Person class

### CODE REVIEW

**Comment by DaoHanLim** [comment]
Includes some tutorial 2 activities because of branching mistakes.

File: src/seedu/addressbook/data/person/Block.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Block.java:3]: No header comment.

File: src/seedu/addressbook/data/person/Block.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	
+	private String blockNum;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Block.java:5]: may be a generic name such as value is sufficient.

File: src/seedu/addressbook/data/person/PostalCode.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	
+	private String postalCodeNum;
+	
+	public PostalCode(String postalCodeNum) {
+		this.postalCodeNum = postalCodeNum;
+	}
+	
+	public String getpostalCodeNum() {
+		return postalCodeNum;
+	}
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/PostalCode.java:13]: Please fix the indentation and change the tab to spaces in Eclipse, see [here](http://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -6,27 +6,53 @@
...
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address must follow format: a/Block, Street, Unit, PostalCode";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_BY = ",";
 
-    public final String value;
-    private boolean isPrivate;
+    public Block block;
+    public Street street;
+    public Unit unit;
+    public PostalCode postalCode;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:19]: Shouldn't it be private?

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -6,27 +6,53 @@
...
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+    	super(address, isPrivate);
+    	if (!isValidAddress(value)){
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:28]: no whitespace before brackets

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -6,27 +6,53 @@
...
         }
-        this.value = trimmedAddress;
+    }
+    
+    /*takes address input, splits into components
+     *  and initializes each class within address*/
+    public void initialiseAddress(String address){
+    	String[] addressComponents = address.split(ADDRESS_SPLIT_BY);
+    	block = new Block(addressComponents[0].trim());
+    	if (addressComponents.length > 1){
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:38]: Should remove magic numbers.

File: src/seedu/addressbook/data/person/Address.java [hunk 7]
```diff
@@ -38,22 +64,12 @@ public static boolean isValidAddress(String test) {
...
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        String line = new String();
+        line+=block.toString();
+        line+=street.toString();
+        line+=unit.toString();
+        line+=postalCode.toString();
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:71]: What is line?? Try to find a better name.

**Comment by jeffryhartanto** [comment]
@DaoHanLim Some comments added. Please close the PR after reading comments. Please submit the PR correctly next time.

**Pull request closed without merging**